File: Codes/2970-count-the-number-of-incremovable-subarrays-i.java
/*
 * @Author: LetMeFly
 * @Date: 2024-07-10 00:21:39
 * @LastEditors: LetMeFly
 * @LastEditTime: 2024-07-10 00:23:41
 */
class Solution {
    private boolean isOk(int[] nums, int l, int r) {
        int last = 0;
        for (int i = 0; i < nums.length; i++) {
            if (l <= i && i <= r) {
                continue;
            }
            if (nums[i] <= last) {
                return false;
            }
            last = nums[i];
        }
        return true;
    }

    public int incremovableSubarrayCount(int[] nums) {
        int ans = 0;
        for (int l = 0; l < nums.length; l++) {
            for (int r = l; r < nums.length; r++) {
                ans += isOk(nums, l, r) ? 1 : 0;
            }
        }
        return ans;
    }
}

File: Codes/2970-count-the-number-of-incremovable-subarrays-i.go
/*
 * @Author: LetMeFly
 * @Date: 2024-07-10 00:24:05
 * @LastEditors: LetMeFly
 * @LastEditTime: 2024-07-10 00:24:10
 */
package main

func isOk(nums []int, l int, r int) int {
    last := 0
    for i := 0; i < len(nums); i++ {
        if l <= i && i <= r {
            continue
        }
        if nums[i] <= last {
            return 0
        }
        last = nums[i]
    }
    return 1
}

func incremovableSubarrayCount(nums []int) int {
    ans := 0
    for l := 0; l < len(nums); l++ {
        for r := l; r < len(nums); r++ {
            ans += isOk(nums, l, r)
        }
    }
    return ans
}